azavea/raster-foundry
# Feature/ak/single date picker

## Overview

This PR adds a single date-picker modal component.

We'll need to revisit our date-pickers and consider rolling our own...trying to piggyback on existing libraries is less than optimal when it comes to both styling and layout.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

It's not super pretty but...

<img width="360" alt="screen shot 2017-05-11 at 5 37 58 pm" src="https://cloud.githubusercontent.com/assets/2442245/25973876/fea341c0-3673-11e7-876a-372461653018.png">

### Notes

This only looks reasonable if you pass `windowClass: 'auto-width-modal'` to the `$uibModal.open` method.

## Testing Instructions

You'll have to modify some code to test this. The `filterPane.controller` file has a call to open the ranged date picker modal ('rfDateRangePickerModal'). Change this to 'rfDatePickerModal'. I'd suggest also logging the value that the modal result is resolved with. Once these changes are made, you can click the date range filter in the imagery browse filter pane to bring up the single date select modal, and on selection, you should see a date (wrapped by moment.js) returned that corresponds to the date you selected.

Closes #1719
